lib/rules/declaration-order.js

```diff
@@ -32,20 +32,29 @@ export default {
     const options = context.options[0] || {};
     const sectionOrder = options.sectionOrder || DEFAULT_SECTION_ORDER;
     const lifecycleOrder = options.lifecycleOrder || DEFAULT_LIFECYCLE_ORDER;
-
     validateSectionOrder(sectionOrder);
-
     return {
+      /** @param {import('vue-eslint-parser').AST.Node} node */
       "Program:exit"(node) {
+        /** @type {import('eslint').SourceCode} */
         const sourceCode = context.getSourceCode();
 
-        // ImportDeclaration은 제외합니다.
-        const nonImportNodes = node.body.filter(
-          (child) => child.type !== "ImportDeclaration",
+        /** @type {import("vue-eslint-parser").AST.VDocumentFragment} */
+        const document = context.sourceCode.parserServices.getDocumentFragment();
+        const setupScripts = document.children.filter(
+            (child) => child.type === "VElement" && child.name === "script" && child.startTag.attributes.find(attr => attr.key.name.toLowerCase() === 'setup')
         );
-        if (nonImportNodes.length === 0) return;
 
-        const nodesWithSection = nonImportNodes.map((child, index) =>
+        const nodes = node.body
+            .filter(
+              (child) => child.type !== "ImportDeclaration",
+            )
+            .filter(node => {
+              const [start, end] = node.range;
+              return setupScripts.some(({range: [scriptStart, scriptEnd]}) => start >= scriptStart && end <= scriptEnd);
+            });
```

Comment:
- If the nodes array is empty, accessing nodes[0].range[0] will result in an undefined error.
- Please add a length check to safely handle this scenario :)

```suggestion
        const nodes = node.body
            .filter(
              (child) => child.type !== "ImportDeclaration",
            )
            .filter(node => {
              const [start, end] = node.range;
              return setupScripts.some(({range: [scriptStart, scriptEnd]}) => start >= scriptStart && end <= scriptEnd);
            });
            
      if (nodes.length === 0) return;          
```